examples/jvm-interop/bridge.c

```diff
@@ -0,0 +1,387 @@
+
+// If the refcount is set to this, that means the allocation is
+// stored in readonly memory in the binary, and we must not
+// attempt to increment or decrement it; if we do, we'll segfault!
+const ssize_t REFCOUNT_READONLY = 0;
+const ssize_t REFCOUNT_ONE = (ssize_t)PTRDIFF_MIN;
+const size_t MASK = (size_t)PTRDIFF_MIN;
+
+// Increment reference count, given a pointer to the first element in a collection.
+// We don't need to check for overflow because in order to overflow a usize worth of refcounts,
+// you'd need to somehow have more pointers in memory than the OS's virtual address space can hold.
+void incref(uint8_t* bytes, uint32_t alignment)
+{
+    ssize_t *refcount_ptr = ((ssize_t *)bytes) - 1;
+    ssize_t refcount = *refcount_ptr;
+
+    if (refcount != REFCOUNT_READONLY) {
+        *refcount_ptr = refcount + 1;
+    }
+}
+
+// Decrement reference count, given a pointer to the first element in a collection.
+// Then call roc_dealloc if nothing is referencing this collection anymore.
+void decref(uint8_t* bytes, uint32_t alignment)
+{
+    if (bytes == NULL) {
+        return;
+    }
+
+    size_t extra_bytes = (sizeof(size_t) >= (size_t)alignment) ? sizeof(size_t) : (size_t)alignment;
+    ssize_t *refcount_ptr = ((ssize_t *)bytes) - 1;
+    ssize_t refcount = *refcount_ptr;
+
+    if (refcount != REFCOUNT_READONLY) {
+        *refcount_ptr = refcount - 1;
+
+        if (refcount == REFCOUNT_ONE) {
+            void *original_allocation = (void *)(refcount_ptr - (extra_bytes - sizeof(size_t)));
+
+            roc_dealloc(original_allocation, alignment);
+        }
+    }
+}
+
+struct RocBytesI32
+{
+    int32_t *bytes;
+    size_t len;
+    size_t capacity;
+};
+
+struct RocBytesI32 init_rocbytes_i32(int32_t *bytes, size_t len)
+{
+    if (len == 0)
+    {
+        struct RocBytesI32 ret = {
+            .len = 0,
+            .bytes = NULL,
+            .capacity = 0,
+        };
+
+        return ret;
+    }
+    else
+    {
+        size_t refcount_size = sizeof(size_t);
+        ssize_t* data = (ssize_t*)roc_alloc(len + refcount_size, alignof(size_t));
+        data[0] = REFCOUNT_ONE;
+        int32_t *new_content = (int32_t *)(data + 1);
+
+        struct RocBytesI32 ret;
+
+        memcpy(new_content, bytes, len * sizeof(int32_t));
+
+        ret.bytes = new_content;
+        ret.len = len;
+        ret.capacity = len;
+
+        return ret;
+    }
+}
+// RocBytesU8 (List U8)
+
+struct RocBytesU8
+{
+    uint8_t *bytes;
+    size_t len;
+    size_t capacity;
+};
+
+struct RocBytesU8 init_rocbytes_u8(uint8_t *bytes, size_t len)
+{
+    if (len == 0)
+    {
+        struct RocBytesU8 ret = {
+            .len = 0,
+            .bytes = NULL,
+            .capacity = 0,
+        };
+
+        return ret;
+    }
+    else
+    {
+        struct RocBytesU8 ret;
+        size_t refcount_size = sizeof(size_t);
+        uint8_t *new_content = ((uint8_t *)roc_alloc(len + refcount_size, alignof(size_t))) + refcount_size;
```

Comment:
This case should also set the refcount.

Reply:
thanks forgot to update it like i32 version